### PR TITLE
AbstractNode#isAttribute should return false if delegate returns Opal.nil

### PR DESCRIPTION
**Pull request by mojavelinux**

### CODE REVIEW

**Comment by Mogztter** [comment]
Yesterday I was reading this part of the code and I was like... hmmm `isAttribute` should return a boolean right.... :thinking:

File: src/asciidoctor-core-api.js [hunk 1]
```diff
@@ -359,10 +359,7 @@ AbstractNode.$$proto.getAttribute = function (name, defaultValue, inherit) {
  */
 AbstractNode.$$proto.isAttribute = function (name, expectedValue, inherit) {
   var value = this['$attr?'](name, expectedValue, inherit);
-  if (value === Opal.nil) {
-    return undefined;
-  }
-  return value;
+  return value === Opal.nil ? false : value;
```
Comment (Mogztter) [inline src/asciidoctor-core-api.js]: Is this behavior consistent with the Asciidoctor core API ? We return a boolean or a value (which could be an instance instance of another type) ?

Comment (mojavelinux) [reply src/asciidoctor-core-api.js]: Technically, core only returns truthy or falsy values, but I think the public-facing API here can be more precise. We should coerce a truthy value to true too. I'll update.

Comment (mojavelinux) [reply src/asciidoctor-core-api.js]: Actually, all we need is a test. It's already returning true and false in all instances as it stands.

Comment (Mogztter) [reply src/asciidoctor-core-api.js]: Perfect then 👌

**Pull request merged**